### ISSUE TITLE
[bug] LR Scheduler updating at wrong times

### ISSUE THREAD

**Issue by ravinkohli** [automl/Auto-PyTorch#211]
Hey,
LR Schedulers dependent on their type should be updated either epoch-wise or batch-wise. Currently, they are only happening batch-wise which is incompatible with CosineAnnealingWarmRestarts, see [here](https://github.com/automl/Auto-PyTorch/blob/ee07c7ef8169fe70945d603b10a263272690397d/autoPyTorch/pipeline/components/training/trainer/base_trainer.py#L321). 

One possible solution to this,
make `{'step'='epochs'}` property of each scheduler, and use this to decide in the base trainer when to make a step. 

I think this is an urgent fix needed.

**Comment by ArlindKadra** [comment]
yeah, I think quite probably the results of the new autopytorch might even be better than the old one now ;)
We should delegate the fix from the cocktail branch as fast as possible

**Comment by ArlindKadra** [comment]
Especially considering the range of `T_0 `is from `0 to 20`
